operator/internal/handlers/lokistack_check_cert_expiry_test.go

```diff
@@ -0,0 +1,459 @@
+			Name:      "my-stack",
+			Namespace: "some-ns",
+		},
+	}
+
+	badRequestErr := apierrors.NewBadRequest("you do not belong here")
+	k.GetStub = func(ctx context.Context, name types.NamespacedName, object client.Object) error {
+		return badRequestErr
+	}
+
+	k.StatusStub = func() client.StatusWriter { return sw }
+
+	err := handlers.CheckCertExpiry(context.TODO(), logger, r, k)
+
+	require.Equal(t, badRequestErr, errors.Unwrap(err))
+
+	// make sure create was NOT called because the Get failed
+	require.Zero(t, k.CreateCallCount())
+}
+
+func TestCheckCertExpiry_WhenGetOptionsReturnsSignerNotFound_DoesNotError(t *testing.T) {
+	sw := &k8sfakes.FakeStatusWriter{}
+	k := &k8sfakes.FakeClient{}
+	r := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "my-stack",
+			Namespace: "some-ns",
+		},
+	}
+
+	stack := lokiv1.LokiStack{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "LokiStack",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-stack",
+			Namespace: "some-ns",
+			UID:       "b23f9a38-9672-499f-8c29-15ede74d3ece",
+		},
+	}
+
+	k.GetStub = func(ctx context.Context, name types.NamespacedName, object client.Object) error {
+		if name.Name == r.Name && name.Namespace == r.Namespace {
+			k.SetClientObject(object, &stack)
+			return nil
+		}
+		return apierrors.NewNotFound(schema.GroupResource{}, "something wasn't found")
+	}
+
+	k.StatusStub = func() client.StatusWriter { return sw }
+
+	err := handlers.CheckCertExpiry(context.TODO(), logger, r, k)
+	require.NoError(t, err)
+
+	// make sure create was NOT called because the Get failed
+	require.Zero(t, k.CreateCallCount())
+}
+
+func TestCheckCertExpiry_WhenGetOptionsReturnsCABUndleNotFound_DoesNotError(t *testing.T) {
+	sw := &k8sfakes.FakeStatusWriter{}
+	k := &k8sfakes.FakeClient{}
+	r := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "my-stack",
+			Namespace: "some-ns",
+		},
+	}
+
+	stack := lokiv1.LokiStack{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "LokiStack",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-stack",
+			Namespace: "some-ns",
+			UID:       "b23f9a38-9672-499f-8c29-15ede74d3ece",
+		},
+	}
+
+	signer := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-stack-signing-ca",
+			Namespace: "some-ns",
+			Labels: map[string]string{
+				"app.kubernetes.io/name":     "loki",
+				"app.kubernetes.io/provider": "openshift",
+				"loki.grafana.com/name":      "my-stack",
+
+				// Add custom label to fake semantic not equal
+				"test": "test",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "loki.grafana.com/v1",
+					Kind:               "LokiStack",
+					Name:               "my-stack",
+					UID:                "b23f9a38-9672-499f-8c29-15ede74d3ece",
+					Controller:         pointer.BoolPtr(true),
+					BlockOwnerDeletion: pointer.BoolPtr(true),
```

Comment:
`BoolPtr` seems to be deprecated:

```suggestion
					Controller:         pointer.Bool(true),
					BlockOwnerDeletion: pointer.Bool(true),
```